fluent-plugin-enhance-k8s-metadata/test/sumologic/kubernetes/test_reader.rb

```diff
@@ -17,7 +17,7 @@ def teardown
   end
 
   def log
-    Fluent::Test::TestLogger.new
+    @test_log ||= Fluent::Test::TestLogger.new
```

Comment:
Why this?

Reply:
If you look at the definition of `log` method before the change, it was returning a new instance of `TestLogger` on every call. This way, if you logged something with `log.error(...)` somewhere in the code, and wanted to see all the logs with `log.logs` later, you would always get an empty collection from `log.logs` because, well, you were calling it on a brand new instance.

What the changed version does, it assigns a new instance of `TestLogger` to class instance member `@test_log` but only if `@test_log` is empty - this is effectively a lazy instantiator for `@test_log` and the `log` method acts as the accessor for this member. This means that the first call to `log` method will create a new `TestLogger` instance and assign it to `@test_log`, and all following calls to `log` method will retrieve the `TestLogger` instance created in the first call to the method.